test/forked/strategies/YearnV3WithGauge.usdc.unit.t.sol

```diff
@@ -0,0 +1,305 @@
+        uint256 gaugeSharesToMint = gauge.previewDeposit(sharesToMint);
+        uint256 gaugeSharesBefore = gauge.balanceOf(address(yearnV3Strategy));
+        // act
+        uint256[] memory slippages = new uint256[](5);
+        slippages[0] = 0;
+        slippages[4] = 1;
+        yearnV3Strategy.exposed_depositToProtocol(assetGroup, Arrays.toArray(toDeposit), slippages);
+
+        // assert
+        uint256 underlyingBalanceOfVaultAfter = vault.totalAssets();
+        uint256 sharesAfter = vault.balanceOf(address(yearnV3Strategy));
+        uint256 gaugeAssetsAfter = gauge.totalAssets();
+        uint256 gaugeSharesAfter = gauge.balanceOf(address(yearnV3Strategy));
+
+        assertEq(underlyingBalanceOfVaultAfter, underlyingBalanceOfVaultBefore + toDeposit, "1");
+        // all shares minted to strategy were put into gauge therefore shares balance is unchanged
+        assertEq(sharesBefore, sharesAfter, "2");
+        assertApproxEqAbs(vault.previewRedeem(sharesToMint), toDeposit, 1, "3");
+
+        //  all vault shares gone into gauge
+        assertEq(gaugeAssetsAfter, gaugeAssetsBefore + sharesToMint, "4");
+        // gauge shares are minted to vault
+        assertEq(gaugeSharesAfter, gaugeSharesToMint + gaugeSharesBefore, "5");
+        assertApproxEqAbs(gauge.previewRedeem(gaugeSharesToMint), sharesToMint, 1, "6");
+    }
+
+    function test_redeemFromProtocol() public {
+        // arrange
+        uint256 mintedShares = 100 * 10 ** 18;
+        uint256 withdrawnShares = 60 * 10 ** 18;
+
+        // - need to deposit into the protocol
+        uint256[] memory slippages = new uint256[](5);
+        slippages[0] = 0;
+        slippages[4] = 1;
+        yearnV3Strategy.exposed_depositToProtocol(assetGroup, Arrays.toArray(toDeposit), slippages);
+        // - normal deposit into protocol would mint SSTs
+        //   which are needed when determining how much to redeem
+        yearnV3Strategy.exposed_mint(mintedShares);
+
+        uint256 underlyingBefore = tokenUnderlying.balanceOf(address(yearnV3Strategy));
+        uint256 sharesBefore = gauge.balanceOf(address(yearnV3Strategy));
+        // act
+        slippages[0] = 1;
+        yearnV3Strategy.exposed_redeemFromProtocol(assetGroup, withdrawnShares, slippages);
+
+        uint256 underlyingAfter = tokenUnderlying.balanceOf(address(yearnV3Strategy));
+        uint256 sharesAfter = gauge.balanceOf(address(yearnV3Strategy));
+
+        // rounding error occurs in both vault and gauge
+        assertApproxEqAbs(underlyingAfter - underlyingBefore, (toDeposit * withdrawnShares) / mintedShares, 2, "1");
+        assertApproxEqAbs(
+            vault.previewRedeem(gauge.previewRedeem(sharesAfter)),
+            (toDeposit * (mintedShares - withdrawnShares)) / mintedShares,
+            2,
+            "2"
+        );
+        assertApproxEqAbs(sharesBefore / sharesAfter, mintedShares / (mintedShares - withdrawnShares), 1, "3");
```

Comment:
This division will loose all precision. This will result in `100 / 40 = 2`.

Reply:
Good point, will adjust that!